src/caf/core/data_structures.py

```diff
@@ -684,8 +870,564 @@ def aggregate(self, segs: list[str]):
+            Whether to do this translation in reverse; by default translations are
+            defined as aggregations, running in reverse will not conserve totals.
+        drop_from: bool = True
+            Whether to remove from_seg from the resulting segmentation.
+        """
+        new_segmentation, lookup = self.segmentation.translate_segment(
+            from_seg, to_seg, reverse=reverse, drop_from=drop_from
+        )
+        if reverse:
+            lookup = lookup.to_frame().set_index(from_seg, append=True).index
+            new_data = self.data.reindex(lookup, level=from_seg)
+            if drop_from:
+                new_data = new_data.droplevel(from_seg)
+        else:
+            try:
+                new_data = self.data.join(lookup).reset_index()
+            except AttributeError:
+                new_data = self.data.to_frame().join(lookup).reset_index()
+            if drop_from:
+                new_data.drop(from_seg, axis=1, inplace=True)
+            new_data = new_data.groupby(new_segmentation.naming_order).sum()
+        return DVector(
+            import_data=new_data,
+            segmentation=new_segmentation,
+            zoning_system=self.zoning_system,
+            time_format=self.time_format,
+            val_col=self.val_col,
+            low_memory=self.low_memory,
+            _bypass_validation=_bypass_validation,
+        )
+
+    def trans_seg_from_lookup(self, lookup: SegConverter, drop_old: bool = False):
+        """
+        Translate segment(s) in self to new segment(s) from a lookup in SegConverter.
+
+        Parameters
+        ----------
+        lookup: SegConverter
+            The name of a lookup in SegConverter
+        drop_old: bool = False
+            Whether to drop old segments from resulting DVector
+        """
+        lookup = SegConverter(lookup).get_conversion()
+        drop_names = lookup.index.names
+        new_names = lookup.columns
+        new_seg = self.segmentation
+        for name in drop_names:
+            if name not in self.segmentation.names:
+                raise ValueError(
+                    f"{name} not in current segmentation so can't" f"be used to convert."
+                )
+            if drop_old:
+                new_seg.remove_segment(name, inplace=True)
+        for name in new_names:
+            new_seg = new_seg.add_segment(SegmentsSuper(name).get_segment())
+
+        new_data = self.data.join(lookup, how="left").reset_index()
+
+        if drop_old:
+            new_data.drop(columns=drop_names, inplace=True)
+        new_data = new_data.groupby(new_seg.naming_order).sum()
+
+        return DVector(
+            import_data=new_data,
+            segmentation=new_seg,
+            zoning_system=self.zoning_system,
+            time_format=self.time_format,
+            low_memory=self.low_memory,
+            val_col=self.val_col,
+            cut_read=True,
         )
 
+    def calc_rmse(self, targets: list[IpfTarget]):
```

Comment:
Return type annotation, I think this returns a single float?
```suggestion
    def calc_rmse(self, targets: list[IpfTarget]) -> float:
```